<a href="https://colab.research.google.com/github/subin100/DataVisualization/blob/main/Quiz7.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##Subin Manandhar
##Data Visualization
##Group Work

###Question: 
How many trips have excatly thirty minutes duration which starts and ends at the same station and subscriber_type is Walk_up ?

Step by Step

In [2]:
from google.colab import auth
auth.authenticate_user()
print('Authenticated')

Authenticated


Lets print first five data to see what type of categories are in the dataset.

In [6]:
%%bigquery --project datavisualization-338803
SELECT *
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
LIMIT 5

trip_id  ... duration_minutes
0  9900289692  ...               39
1  9900285987  ...               31
2  9900285989  ...               31
3  9900285991  ...               30
4  9900286140  ...               19

[5 rows x 9 columns]

From this dataset we will be using subsciber_type, start_station_name,end_station_name and duration_minutes in this project.

To begin with, lets find out how many trips have duration of excatly 30 mins.

In [18]:
%%bigquery --project datavisualization-338803
SELECT  COUNTIF(duration_minutes=30) as Thirty_Minutes_duration
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`

Thirty_Minutes_duration
0                    12236

So, 12236 trips have 30 minutes duration.


Now to complicate this problem and trim this data even more, We are going to find out how many trips have thirty minutes. duration which starts and ends at the same station.

In [11]:
%%bigquery --project datavisualization-338803
SELECT COUNT(*) as Thirty_MinuteS_Trips
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
WHERE start_station_name = end_station_name AND duration_minutes =30


Thirty_MinuteS_Trips
0                  2476

So this has excluded many trips. only 2476 trips starts and ends at same station which have 30 minutes duration.

Lets trim this even more and bring subscirber type to this equation.
Lets find out how many trips have thirty minutes duration which starts and ends at the same station and subscriber_type is Walk_up

In [17]:
%%bigquery --project datavisualization-338803
SELECT COUNT(*) as Thirty_MinuteS_Trips_AND_WALK_UP
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
WHERE duration_minutes =30 AND subscriber_type ="Walk Up" AND start_station_name = end_station_name

Thirty_MinuteS_Trips_AND_WALK_UP
0                               928

From this output, We found out that less than 1000 trips fall into the category we are looking for.